File: VB/WebSite/ReportPreview.ascx.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.XtraScheduler.Reporting
Imports DevExpress.Web.ASPxScheduler.Reporting

Partial Public Class ReportPreview
	Inherits System.Web.UI.UserControl
	Private reportTemplateFileName_Renamed As String
	Private controlAdapter_Renamed As ASPxSchedulerControlPrintAdapter

	Public Property ReportTemplateFileName() As String
		Get
			Return reportTemplateFileName_Renamed
		End Get
		Set(ByVal value As String)
			reportTemplateFileName_Renamed = value
		End Set
	End Property

	Public Property ControlAdapter() As ASPxSchedulerControlPrintAdapter
		Get
			Return controlAdapter_Renamed
		End Get
		Set(ByVal value As ASPxSchedulerControlPrintAdapter)
			controlAdapter_Renamed = value
		End Set
	End Property

	Protected Sub Page_Load(ByVal sender As Object, ByVal e As EventArgs)
		Dim rpt As New XtraSchedulerReport()
		rpt.LoadLayout(Page.MapPath("~/App_Data/" & ReportTemplateFileName))
		rpt.SchedulerAdapter = ControlAdapter.SchedulerAdapter
		ReportViewer1.Report = rpt
	End Sub
End Class

File: CS/WebSite/ReportPreview.ascx.cs
using System;
using DevExpress.XtraScheduler.Reporting;
using DevExpress.Web.ASPxScheduler.Reporting;

public partial class ReportPreview : System.Web.UI.UserControl {
    string reportTemplateFileName;
    ASPxSchedulerControlPrintAdapter controlAdapter;

    public string ReportTemplateFileName { 
        get { return reportTemplateFileName; } 
        set { reportTemplateFileName = value; } 
    }

    public ASPxSchedulerControlPrintAdapter ControlAdapter { 
        get { return controlAdapter; } 
        set { controlAdapter = value; } 
    }

    protected void Page_Load(object sender, EventArgs e) {
        XtraSchedulerReport rpt = new XtraSchedulerReport();
        rpt.LoadLayout(Page.MapPath("~/App_Data/" + ReportTemplateFileName));
        rpt.SchedulerAdapter = ControlAdapter.SchedulerAdapter;
        ReportViewer1.Report = rpt;
    }
}